File: api/HrManagementSystem/Features/Platform/Notifications/Entities/Notification.cs
namespace HrManagementSystem.Features.Platform.Notifications.Entities;

public enum NotificationSeverity
{
    Info = 1,
    Success = 2,
    Warning = 3,
    Critical = 4
}

public sealed class Notification
{
    public long Id { get; set; }
    public string RecipientUserId { get; set; } = string.Empty;
    public string? ActorUserId { get; set; }
    public string RequiredPermission { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string EventType { get; set; } = string.Empty;
    public NotificationSeverity Severity { get; set; }
    public string TitleKey { get; set; } = string.Empty;
    public string MessageKey { get; set; } = string.Empty;
    public string ParametersJson { get; set; } = "{}";
    public string? EntityType { get; set; }
    public string? EntityId { get; set; }
    public string? ActionUrl { get; set; }
    public Guid CorrelationId { get; set; }
    public string? DeduplicationKey { get; set; }
    public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
    public DateTime? ReadOn { get; set; }
    public DateTime? DismissedOn { get; set; }
    public DateTime? ExpiresOn { get; set; }
    public DateTime? DeliveredOn { get; set; }
    public int DeliveryAttempts { get; set; }
    public DateTime? NextDeliveryAttemptOn { get; set; }
    public DateTime? DeliveryClaimedUntil { get; set; }
    public string? LastDeliveryError { get; set; }

    public ApplicationUser RecipientUser { get; set; } = default!;
    public ApplicationUser? ActorUser { get; set; }
}

File: mobile-react/src/features/notifications/types/notification.ts
import type { PageMetadata } from '@/src/core/api';

export type NotificationSeverity = 1 | 2 | 3 | 4 | 'Info' | 'Success' | 'Warning' | 'Critical';
export type NotificationReadStatus = 0 | 1 | 2;
export type NotificationFilter = 'all' | 'unread' | 'read';

export interface AppNotification {
  id: number;
  category: string;
  eventType: string;
  severity: NotificationSeverity;
  titleKey: string;
  messageKey: string;
  parameters: Record<string, string>;
  entityType: string | null;
  entityId: string | null;
  actionUrl: string | null;
  actorUserId: string | null;
  correlationId: string;
  createdOn: string;
  readOn: string | null;
  expiresOn: string | null;
}

export interface NotificationQuery {
  pageNumber?: number;
  pageSize?: number;
  status?: NotificationReadStatus;
  category?: string;
  severity?: Exclude<NotificationSeverity, string>;
  columnName?: 'CreatedOn' | 'ReadOn' | 'Severity' | 'Category';
  sortDirection?: 'ASC' | 'DESC';
}

export interface NotificationPageResponse {
  items: AppNotification[];
  metaData: PageMetadata;
}
